torch/_higher_order_ops/cast_symbool_to_symint.py

```diff
@@ -0,0 +1,69 @@
+import sympy
+
+import torch
+
+from torch._C import DispatchKey
+from torch._ops import HigherOrderOperator
+from torch._subclasses.fake_tensor import FakeTensorMode
+
+from torch.fx.experimental.proxy_tensor import ProxyTorchDispatchMode, track_tensor_tree
+
+cast_symbool_to_symint = HigherOrderOperator("cast_symbool_to_symint")
+
+
+def convert_symbool_to_int_with_hint(maybe_symbool):
```

Comment:
I find this name fairly uninformative as to what this does (and there is no docblock), also this feels like it is in the "wrong" place as it isn't used by the hop